### PR TITLE
Small cleanup of IRingRange.

### PR DESCRIPTION
**Pull request by gabikliot**
Small cleanup of IRingRange, before introducing sequence number to be embedded in the IRingRange.

Just moved the code into a separate file and also made sure RangeFactory is used everywhere instead of constructor.

### CODE REVIEW

File: src/Orleans/Runtime/IRingRange.cs [hunk 1]
```diff
@@ -0,0 +1,356 @@
...
+        {
+            return new SingleRange(begin, end);
+        }
+
+        public static IRingRange CreateRange(List<IRingRange> inRanges)
+        {
+            return new GeneralMultiRange(inRanges);
+        }
+
+        public static EquallyDevidedMultiRange CreateEquallyDevidedMultiRange(IRingRange range, int numSubRanges)
```
Comment (dVakulen) [inline src/Orleans/Runtime/IRingRange.cs]: Nit: 'Divided' instead of 'Devided'.

**Comment by sergeybykov** [comment]
>made sure RangeFactory is used everywhere instead of constructor.

Would it makes even more sense to make the rage class private nested classes of `RangeFactory`, so that they aren't even directly accessible?

Also, `Runtime\IRingRange.cs` looks somewhat misleading with several interfaces and classes defined in it. Maybe put them into individual source files under `Orleans\Runtime\Ring\` instead?

**Comment by gabikliot** [comment]
I renamed the file to RingRange.
We can't make the classes  private nested, since SingleRange has to be public, since we need access to its Begin/End in reminders.

**Comment by sergeybykov** [comment]
>public interface RingRange

Looks like it auto-rename the interface.

**Comment by gabikliot** [comment]
Fixed.

**Comment by sergeybykov** [comment]
@dotnet-bot Test this please.

**Comment by gabikliot** [comment]
Thank you @sergeybykov .

**Comment by sergeybykov** [comment]
Thank you, Gabi, for doing it.

**Pull request merged**